### PR TITLE
Revendor hcsshim to v0.4.1

### PR DESCRIPTION
**Pull request by darstahl**
**\- What I did**

Revendor hcsshim to v0.4.0. This requires changing some error checking to use the new methods to check multiple error types which mean the same thing.

The new update paves the way to support docker stats on Windows.

Signed-off-by: Darren Stahl darst@microsoft.com

### CODE REVIEW

**Comment by thaJeztah** [comment]
> The new update paves the way to support docker stats.

nice!

File: libcontainerd/client_windows.go [hunk 1]
```diff
@@ -294,13 +294,13 @@ func (clnt *client) Signal(containerID string, sig int) error {
...
 		if err := cont.hcsContainer.Terminate(); err != nil {
-			if err != hcsshim.ErrVmcomputeOperationPending {
+			if !hcsshim.IsPending(err) {
 				logrus.Errorf("libcontainerd: failed to terminate %s - %q", containerID, err)
 			}
 		}
 	} else {
 		// Terminate Process
-		if err := cont.hcsProcess.Kill(); err != nil {
+		if err := cont.hcsProcess.Kill(); err != nil && !hcsshim.IsAlreadyStopped(err) {
```
Comment (thaJeztah) [inline libcontainerd/client_windows.go]: Would be nice to make `IsAlreadyStopped()` check for `nil` (similar to https://golang.org/src/os/error_unix.go?h=isNotExist#L25), then checking for nil can be skipped here, and simply

```golang
if err := cont.hcsProcess.Kill(); !hcsshim.IsAlreadyStopped(err) {
```

Comment (darrenstahlmsft) [inline libcontainerd/client_windows.go]: I think IsNotExist also needs the nil check. The return of false would indicate that an error was returned that is not caused by a missing file, but it still might be caused by something else.

Comment (thaJeztah) [inline libcontainerd/client_windows.go]: oh, you're absolutely right, got bit by the double negative. Ignore me. Also, it already handles the `nil` value, so all bogus. I shouldn't be doing late-night code reading 👍

**Comment by darrenstahlmsft** [comment]
CI error looks like it might be legit, very curious why it fails to load an image in CI but tests all pass locally. I'll update this PR tomorrow.

**Comment by vdemeester** [comment]
It's green, LGTM :angel:

**Comment by thaJeztah** [comment]
LGTM, thanks!

**Pull request merged**